runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/external/PythonExecutorLifecycleManagerImpl.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright © 2014-2017 EntIT Software LLC, a Micro Focus company (L.P.)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudslang.runtime.impl.python.external;
+
+import io.cloudslang.runtime.api.python.PythonExecutorConfigurationDataService;
+import io.cloudslang.runtime.api.python.PythonExecutorLifecycleManager;
+import io.cloudslang.runtime.api.python.entities.PythonExecutorDetails;
+import io.cloudslang.runtime.api.python.enums.PythonStrategy;
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static io.cloudslang.runtime.api.python.enums.PythonStrategy.PYTHON_EXECUTOR;
+import static io.cloudslang.runtime.api.python.enums.PythonStrategy.getPythonStrategy;
+import static java.io.File.separator;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+import static org.jboss.resteasy.util.HttpHeaderNames.AUTHORIZATION;
+import static org.jboss.resteasy.util.HttpHeaderNames.CONTENT_TYPE;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON;
+
+@Service("pythonExecutorLifecycleManager")
+public class PythonExecutorLifecycleManagerImpl implements PythonExecutorLifecycleManager {
+
+    private static final Logger logger = LogManager.getLogger(PythonExecutorLifecycleManagerImpl.class);
+    private static final PythonStrategy PYTHON_EVALUATOR = getPythonStrategy(System.getProperty("python.expressionsEval"), PYTHON_EXECUTOR);
+    private static final String EXTERNAL_PYTHON_EXECUTOR_STOP_PATH = "/rest/v1/stop";
+    private static final String EXTERNAL_PYTHON_EXECUTOR_HEALTH_PATH = "/rest/v1/health";
+    private static String PYTHON_EXECUTOR_PORT;
```

Comment:
No need to store this info in static fields, simply use pythonExecutorConfigurationDataService.getPythonExecutorConfiguration() where you need the fields. That already hits an in memory object.